ykrt/src/trace/hwt/mapper.rs

```diff
@@ -135,95 +217,3 @@ fn map_block(block: &hwtracer::Block) -> Vec<Option<TraceAction>> {
     }
     ret
 }
-
-/// Map the *machine* blocks of the specified trace into LLVM IR blocks.
-///
-/// Each entry in the returned trace is either a "mapped block" identifying a successfully
-/// mapped LLVM IR block, or an unsuccessfully mapped "unmappable block" (an unknown region of
-/// code spanning at least one machine block).
-///
-/// In the returned trace, unmappable blocks never appear consecutively.
-///
-/// The returned trace will always start with a mapped block (the unmappable prefix of the
-/// foreign "turn on tracing" routine is omitted).
-pub fn map_trace(trace: Box<dyn Trace>) -> Result<Vec<TraceAction>, Box<dyn Error>> {
-    let mut ret: Vec<TraceAction> = Vec::new();
-
-    let mut trace_iter = trace.iter_blocks();
-    // The first block contains the control point so we need to remove that.
-    match trace_iter.next() {
-        Some(Ok(x)) => {
-            // As a rough proxy for "check that we removed only the thing we want to remove",
-            // we know that the control point will be contained in a single mappable block.
-            assert!(matches!(
-                map_block(&x).as_slice(),
-                &[Some(TraceAction::MappedAOTBlock { .. })]
-            ));
-        }
-        _ => unreachable!(),
-    }
-
-    for (i, block) in trace_iter.enumerate() {
-        if i > crate::mt::DEFAULT_TRACE_TOO_LONG {
-            return Err("Trace too long".into());
-        }
-        let block = block?;
-        let irblocks = map_block(&block);
-        if irblocks.is_empty() {
-            // The block is unmappable. Insert a IRBlock that indicates this, but only if the
-            // trace isn't empty (we never report the leading unmappable code in a trace). We
-            // also take care to collapse consecutive unmappable blocks into one.
-            if let Some(last) = ret.last_mut() {
-                match last {
-                    TraceAction::MappedAOTBlock { .. } => {
-                        ret.push(TraceAction::new_unmappable_block());
-                    }
-                    TraceAction::UnmappableBlock => (),
-                    TraceAction::Promotion => todo!(),
-                }
-            }
-        } else {
-            for irblock in irblocks.into_iter() {
-                if let Some(irb) = irblock {
-                    match ret.last() {
-                        Some(last) if &irb != last => ret.push(irb),
-                        Some(_) => {
-                            // The `BlockDisambiguate` pass in ykllvm ensures that no
```

Comment:
We seem to have lost this comment in the churn. It's an important one worth keeping IMHO.